### DISCUSSION TITLE
How to save iframe's content in standalone mode

### DISCUSSION THREAD

**Discussion by dontfireme** [gildas-lormeau/SingleFile#1055] [Q&A]
I wanted to use SingleFile from JS console to archive some pages. I noticed that the standalone mode (copy-pastying SingleFile source into browser's console) cannot access some of the iframes due to the same-origin policy.

This is how I use it in the browser:

```
const { content, title, filename } = await window.singlefile.getPageData({
  removeImports: true,
  removeScripts: true,
  removeAudioSrc: true,
  removeVideoSrc: true  
});
```

essentially this piece of code is throwing:

![image](https://user-images.githubusercontent.com/31350083/202263268-8c70a27f-a211-4da3-94b3-4426d0cf25d3.png)

Everything works well when using extension, so, my question is what kind of magic the extension does that makes it work? 🤔 

Thank you for the awesome package!

**Comment by gildas-lormeau** [comment]
The code of SingleFile is injected in all frames when running into an extension. It relies on `postMessage` to pass data from frames to the main document when they're not from the same origin. See, https://github.com/gildas-lormeau/single-file-core/blob/6444c2ed3efea177d9ab0eb549944f03f535a425/processors/frame-tree/content/content-frame-tree.js#L263-L288
